gui/src/ts/components/kg/node/KgNodeTable.tsx

```diff
@@ -25,66 +15,158 @@ const showListAsColumn = (list: string[]) =>
     </React.Fragment>
   ));
 
+const columns: MUIDataTableColumn[] = [
+  {
+    name: "#",
+    options: {
+      customBodyRender(_, tableMeta) {
+        return (
+          tableMeta.tableState.page * tableMeta.tableState.rowsPerPage +
+          tableMeta.rowIndex +
+          1
+        );
+      },
+    },
+  },
+  {
+    name: "id",
+    options: {
+      display: "false",
+    },
+  },
+  {
+    name: "label",
+    label: "Label",
+    options: {
+      customBodyRender(_, tableMeta) {
+        const nodeRowData = (tableMeta.tableData[
+          tableMeta.rowIndex
+        ] as unknown) as (string | undefined)[];
+
+        return (
+          <KgNodeLink
+            node={{
+              id: nodeRowData[getPropertyColumnIndex("id")]!,
+              label: nodeRowData[getPropertyColumnIndex("label")] || null,
+              pos: nodeRowData[getPropertyColumnIndex("pos")] || null,
+            }}
+          />
+        );
+      },
+    },
+  },
+  {
+    name: "aliases",
+    label: "Aliases",
+    options: {
+      customBodyRender(aliases) {
+        return aliases ? showListAsColumn(aliases as string[]) : null;
+      },
+    },
+  },
+  {
+    name: "datasource",
+    label: "Datasources",
+    options: {
+      customBodyRender(datasources) {
+        return datasources
+          ? (datasources as string).split(",").map((datasource) => (
+              <React.Fragment key={datasource}>
+                <KgDatasourceLink datasource={datasource} />
+                <br />
+              </React.Fragment>
+            ))
+          : null;
+      },
+    },
+  },
+  {
+    name: "other",
+    label: "Other",
+  },
+  {
+    name: "pos",
+    label: "Pos",
+    options: {
+      customBodyRender(pos) {
+        return pos ? showListAsColumn((pos as string).split(",")) : null;
+      },
+    },
+  },
+];
+
+const getPropertyColumnIndex = (
+  property: Exclude<keyof KgNode, "__typename">
+) => {
+  return columns.findIndex(
+    (col) => typeof col !== "string" && col.name === property
+  );
+};
+
 export const KgNodeTable: React.FunctionComponent<{
+  title: string;
   nodes: KgNode[];
   rowsPerPage: number;
   count: number;
   page: number;
   onChangePage: (newPage: number) => void;
   onChangeRowsPerPage: (newRowsPerPage: number) => void;
-}> = ({nodes, rowsPerPage, count, page, onChangePage, onChangeRowsPerPage}) => {
+}> = ({
+  title,
+  nodes,
+  rowsPerPage,
+  count,
+  page,
+  onChangePage,
+  onChangeRowsPerPage,
+}) => {
+  // https://github.com/gregnb/mui-datatables/issues/756
+  // Since the MUIDataTable has its own state, it ignores passed in values
+  // for page and rowsPerPage
+  // Workaround by checking the table state via ref and updating
+  // with internal api
+  const tableCallbackRef = (table: any) => {
```

Comment:
You need this in order to keep the query string updated?

Reply:
This is used to update the table state if the query string updates (e.g. pressing back button or directly going to url).